File: src/test/scala/com/exasol/cloudetl/kafka/ittests/docker/SchemaRegistryContainer.scala
package com.exasol.cloudetl.kafka

import org.testcontainers.containers.GenericContainer
import org.testcontainers.utility.DockerImageName

/**
 * A Schema Registry container that exposes a fixed port.
 */
class SchemaRegistryContainer(image: DockerImageName, fixedPort: Int = 8081) extends GenericContainer(image) {

  override def configure(): Unit = {
    super.configure()
    addFixedExposedPort(fixedPort, fixedPort)
  }

}

File: src/test/java/com/exasol/cloudetl/kafka/SchemaRegistryContainer.java
package com.exasol.cloudetl.kafka;

import org.testcontainers.containers.GenericContainer;
import org.testcontainers.utility.DockerImageName;

class SchemaRegistryContainer extends GenericContainer<SchemaRegistryContainer> {
    private final int fixedPort;

    SchemaRegistryContainer(final DockerImageName image, final int fixedPort) {
        super(image);
        this.fixedPort = fixedPort;
    }

    @Override
    protected void configure() {
        super.configure();
        addFixedExposedPort(this.fixedPort, this.fixedPort);
    }
}
